tests/jballerina-unit-test/src/test/resources/test-src/annotations/deprecation_annotation.bal

```diff
@@ -53,3 +53,149 @@ public function func3() {
 public function func4() {
     func3();
 }
+
+type Annot record {
+    string foo;
+    int bar?;
+};
+
+# Annotation for record 'Annot'
+#
+# # Deprecated
+@deprecated
+public annotation Annot v1 on type;
+@deprecated
+public annotation Annot[] v2 on object type;
+@deprecated
+public annotation map<int> v4 on object function, function;
+
+string strValue = "v1 value";
+
+@v1 {
+    foo: strValue,
+    bar: 1
+}
+public type T1 record {
+    string name;
+};
+
+T1 a = { name: "John" };
+
+function testAnnotationDeprecation() {
+    typedesc<any> t = typeof a;
+    Annot? annot1 = t.@v1;
+    Annot[]? annot2 = t.@v2;
+}
+
+@v1 {
+    foo: strValue
+}
+@v2 {
+    foo: "v2 value 1"
+}
+@v2 {
+    foo: "v2 value 2"
+}
+type T2 object {
+    string name = "ballerina";
+
+    @v4 {
+        val: 42
+    }
+    public function objMethod() {
+    }
+};
+
+@v4 {
+        val: 42
+}
+public function deprecated_annotation_func() {
+}
+
+# Test function doc
+# + x - first integer
+# + y - second integer
+# + z - third integer
+# + return - Returns the sum
+# # Deprecated parameters
+# + x - deprecated parameter
+# + y - deprecated parameter
+# + z - deprecated parameter
+# # Deprecated
+# function is deprecated
+@deprecated
+function add1(@deprecated public int x, @deprecated int y, @deprecated int z) returns int {
+    return x + y + z;
+}
+
+@deprecated
+function add2(@deprecated public int x, @deprecated int y, @deprecated int z) returns int {
+    return x + y + z;
+}
+
+const string CONST3 = "CONST3";
+const string CONST4 = "CONST4";
+
+public type TYPE1 CONST3|CONST4;
+
+@deprecated
+public type Object1 object {
+
+    @deprecated
+    public string fieldOne = "Foo";
+    @deprecated
+    TYPE1 t = CONST3;
+    public string fieldTwo = "";
+
+    public function doThatOnObject(string paramOne, TYPE1 t) {
+        self.fieldOne = paramOne;
+        self.t = t;
+    }
+};
+
+# The `Object2` is a user-defined object.
+#
+# + fieldOne - This is the description of the `Object2`'s `fieldOne` field.
+# + fieldTwo - This is the description of the `Object2`'s `fieldTwo` field.
+# # Deprecated parameters
```

Comment:
We have fields in an object, not parameters. I think we should allow documenting like this **only for functions**.

For objects, if someone wants to add deprecated docs for a field, shouldn't the following (documenting at field level) work and be enough?

```ballerina
# The `Object2` is a user-defined object.
# 
# + fieldTwo - This is the description of the `Object2`'s `fieldTwo` field.
public type Object2 object {

    # This is the description of the `Object2`'s `fieldOne` field.
    # # Deprecated
    # deprecated docs
    @deprecated
    public string fieldOne = "Foo";

    # This is the description of the `Object2`'s `t` field.
    # # Deprecated
    # deprecated docs
    @deprecated
    TYPE1 t = CONST3;

    public string fieldTwo = "";

    public function doThatOnObject(string paramOne, TYPE1 t) {
        self.fieldOne = paramOne;
        self.t = t;
    }
};

const string CONST3 = "CONST3";
const string CONST4 = "CONST4";

public type TYPE1 CONST3|CONST4;
```

Currently says
```cmd
error: .::xyz.bal:6:5: constructs annotated as '@deprecated' must have 'Deprecated' documentation
error: .::xyz.bal:12:5: constructs annotated as '@deprecated' must have 'Deprecated' documentation
```

Reply:
It is not mentioned in the spec. That why we choose one way to document the separation in parameters

Reply:
For function parameters yeah, because we don't have any other way to document them. 

For object fields, we have a straightforward way to do this, and we'll probably not support documenting fields at type level anyway in a future release (since they can now be documented at field level.). WDYT?

Looping in @hasithaa, @pubudu91 too.

Reply:
Yeah, the only reason we needed a new way for params is because documentation is not allowed on params.

Reply:
Fixed